UI-side Compositing: PDF scrollbars are not draggable and don't grow when hovered

rr-codes opened the pull request:
#### 6d3572a224f0e70a963e9d00293e2c9562372ff8
<pre>
UI-side Compositing: PDF scrollbars are not draggable and don&apos;t grow when hovered
<a href="https://bugs.webkit.org/show_bug.cgi?id=254363">https://bugs.webkit.org/show_bug.cgi?id=254363</a>
rdar://105908418

Reviewed by Tim Horton.

For some reason, when UI-side compositing is enabled, the scrollbar&apos;s layer&apos;s backing store
is larger than the layer itself when viewing a zoomed-in PDF. This causes the scrollbar&apos;s
appearance&apos;s y-value to be offset by some amount.

This PR works around this by clearing the contents of the layer so that CoreAnimation creates
a new backing store whenever the frame&apos;s change, which forces the new backing store to have the
correct size.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::updateScrollbars):

Canonical link: <a href="https://commits.webkit.org/262043@main">https://commits.webkit.org/262043@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/562bfa8fdf2200316fe266cd062203f3c889baba

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/351 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/367 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/383 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/356 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/332 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/409 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/402 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/575 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/359 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/347 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/342 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/384 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/332 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/312 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/349 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/336 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/323 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/304 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/344 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/90 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/341 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 reviewed (approved):
I still wish we knew what was going on here.

rr-codes committed:
UI-side Compositing: PDF scrollbars are not draggable and don't grow when hovered
https://bugs.webkit.org/show_bug.cgi?id=254363
rdar://105908418

Reviewed by Tim Horton.

For some reason, when UI-side compositing is enabled, the scrollbar's layer's backing store
is larger than the layer itself when viewing a zoomed-in PDF. This causes the scrollbar's
appearance's y-value to be offset by some amount.

This PR works around this by clearing the contents of the layer so that CoreAnimation creates
a new backing store whenever the frame's change, which forces the new backing store to have the
correct size.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::updateScrollbars):

Canonical link: https://commits.webkit.org/262043@main
